### PR TITLE
K8SPXC-765: add ConfigMaps deletion for custom configurations

### PR DESCRIPTION
**Pull request by ollevche**
[K8SPXC-765](https://jira.percona.com/browse/K8SPXC-765) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPXC-765

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| storage | passed |
| scheduled-backup | passed |
| big-data | passed |
| haproxy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| init-deploy | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| upgrade-proxysql | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| smart-update | passed |
| affinity | passed |
| one-pod | failed |
| self-healing-advanced-chaos | passed |
| pitr | failed |
| auto-tuning | passed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| upgrade-consistency | passed |
| operator-self-healing-chaos | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/889/commits/9028048416764d320f29cc1b28c948081b1dcab0
image: `perconalab/percona-xtradb-cluster-operator:PR-889-90280484`

**Review by delgod — APPROVED** [review]

File: pkg/controller/pxc/controller.go [hunk 1]
```diff
@@ -844,10 +846,16 @@ func (r *ReconcilePerconaXtraDBCluster) reconcileConfigMap(cr *api.PerconaXtraDB
 		if err != nil {
 			return errors.Wrap(err, "pxc config map")
 		}
+	} else {
+		if err := deleteConfigMapIfExists(r.client, cr, pxcConfigName); err != nil {
+			return errors.Wrap(err, "delete pxc config map")
+		}
 	}
 
+	var proxysqlConfigName = ls["app.kubernetes.io/instance"] + "-proxysql"
```
Comment (qJkee) [inline pkg/controller/pxc/controller.go:855]: please, do not use `var`, use `:=` operator

Comment (ollevche) [reply pkg/controller/pxc/controller.go]: 36ed173

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| recreate | passed |
| haproxy | passed |
| scheduled-backup | passed |
| storage | passed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| affinity | passed |
| self-healing-advanced | passed |
| one-pod | passed |
| self-healing-advanced-chaos | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| users | passed |
| upgrade-consistency | passed |
| operator-self-healing-chaos | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/889/commits/36ed173e22b9ea0d985343fabeab9182217a1666
image: `perconalab/percona-xtradb-cluster-operator:PR-889-36ed173e`

File: pkg/controller/pxc/controller.go [hunk 2]
```diff
@@ -870,22 +884,34 @@ func (r *ReconcilePerconaXtraDBCluster) reconcileConfigMap(cr *api.PerconaXtraDB
...
+		logCollectorConfigName := ls["app.kubernetes.io/instance"] + "-logcollector"
+
+		if cr.Spec.LogCollector != nil && cr.Spec.LogCollector.Configuration != "" {
+			configMap := config.NewConfigMap(cr, logCollectorConfigName, "fluentbit_custom.conf", cr.Spec.LogCollector.Configuration)
+			err := setControllerReference(cr, configMap, r.scheme)
 			if err != nil {
-				return errors.Wrap(err, "update ConfigMap LogCollector")
+				return errors.Wrap(err, "set controller ref LogCollector")
+			}
+			err = r.client.Create(context.TODO(), configMap)
```
Comment (qJkee) [inline pkg/controller/pxc/controller.go:902]: pls call `createOrUpdateConfigmap`

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/889/commits/fe48de3557d2877961bddb956d4ad60b664d28c2
image: `perconalab/percona-xtradb-cluster-operator:PR-889-fe48de35`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| upgrade-haproxy | passed |
| self-healing | passed |
| upgrade-proxysql | passed |
| self-healing-chaos | passed |
| smart-update | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| operator-self-healing-chaos | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/889/commits/daa4494103c3c020131faf690083e2ade2ce1c50
image: `perconalab/percona-xtradb-cluster-operator:PR-889-daa44941`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| haproxy | passed |
| big-data | passed |
| scheduled-backup | passed |
| upgrade-haproxy | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| smart-update | passed |
| self-healing-advanced | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| self-healing-advanced-chaos | passed |
| proxysql-sidecar-res-limits | failed |
| demand-backup | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/889/commits/fe48de3557d2877961bddb956d4ad60b664d28c2
image: `perconalab/percona-xtradb-cluster-operator:PR-889-fe48de35`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| upgrade-haproxy | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| scaling | passed |
| haproxy | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| security-context | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| big-data | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/889/commits/fe48de3557d2877961bddb956d4ad60b664d28c2
image: `perconalab/percona-xtradb-cluster-operator:PR-889-fe48de35`

**Pull request merged**